src/core/hyprlock.cpp

```diff
@@ -393,6 +390,9 @@ void CHyprlock::run() {
 
     acquireSessionLock();
 
+    // gather info about monitors
+    wl_display_roundtrip(m_sWaylandState.display);
```

Comment:
I dont think this is correct

Reply:
Yeah, I just noticed, that the second instance of hyprlock now segfaults. Initially, I moved the `acqureSessionLock` call onto line 368 (i.e. after the condition, but before `wl_display_roundtrip`), which also resolves the problem, though this introduces a brief red flashing screen, which does not happen when `wl_display_roundtrip` is called after the creation of the renderer.